README.rst

```diff
@@ -122,10 +123,124 @@ through *mapping*. Currently only **XML mapping** supported:
-field. (We use the Symfony `PropertyAccess
-<http://symfony.com/doc/current/components/property_access/index.html>`_
-component, so it works on properties and methods alike).
+an index named ``product`` using the ``locale`` determined from the locale
+field and the ID obtained from the objects ``id``
+field.
+
+Mapping elements
+~~~~~~~~~~~~~~~~
+
+The possible mappings are:
+
+- ``index``: Name of the index in which to insert the record
+- ``title``: Title to use in search results
+- ``description``: A description for the search result
+- ``url``: The URL to which the search resolt should link to
+- ``image``: An image to associate with the search result
+- ``fields``: List of ``<field />`` elements detailing which fields should be
+  indexed (i.e. used when finding search results).
+
+Each mapping can use either a ``property`` attribute or an ``expr`` attribute.
+These attributes determine how the value is retrieved. ``property`` will use
+the Symfony `PropertyAccess`_ component, and ``expr`` will use
+`ExpressionLanguage`_.
+
+PropertyAccess allows you to access properties of an object by path, e.g.
+``title``, or ``parent.title``. The expression allows you to build expressions
+which can be evaluated, e.g. ``'/this/is/' ~ object.id ~ '/a/path'``.
+
+Fields
+~~~~~~
+
+Fields dictate which fields are indexed in the underlying search engine.
+
+Mapping:
+
+- ``name``: Field name
+- ``property``: Object property to map the field
+- ``expr``: Mutually exclusive with ``property``
+
+Types:
+
+- ``string``: Store as a string
+
+Expression Language
+~~~~~~~~~~~~~~~~~~~
+
+The MassiveSearchBundle includes its own flavor of the Symfony expression
+language.
+
+Functions:
+
+- ``join``: Maps to the ``implode`` function in PHP. e.g. ``join(',', ["one",
+  "two"])`` equals ``"one,two"``
+- ``map``: Maps to the ``array_map`` function in PHP. e.g. ``map([1, 2, 3],
+  'el + 1')`` equals ``array(2, 3, 4)``.
+
+Contexts
+~~~~~~~~
+
+Sometimes you will require different mappings based on the context of the web
+application. For example, an Article may have one URL in the front office, and
+another in the backoffice (i.e. for viewing and editing respectively).
+
+MassiveSearchBundle solves this with ``contexts``. A ``context`` allows you
+to map additional indexes:
+
+.. code-block:: xml
+
+    <!-- /path/to/YourBundle/Resources/config/massive-search/Product.xml -->
+    <massive-search-mapping xmlns="http://massive.io/schema/dic/massive-search-mapping">
+
+        <mapping class="Massive\Bundle\SearchBundle\Tests\Resources\TestBundle\Entity\Product">
+            <index name="product" />
+            <id property="id" />
+            <url expr="'/path/to/' ~ article.title'" />
+            <fields>
+                <field name="title" type="string" />
+                <field name="body" type="string" />
+            </fields>
+
+            <context name="admin">
+                <url exp="'/admin/edit/article/' ~ object.id" />
```

Comment:
I am still not a fan of this :-/ It couples the backend too much to the frontend... @3rfan Do you build now this URL building stuff we have talked about? Would make this somehow obsolete...